### DISCUSSION TITLE
Source for "Custom Palette"

### DISCUSSION THREAD

**Discussion by lumosminima** [radix-ui/colors#59] [Ideas]
I saw that there is a new [https://www.radix-ui.com/colors/custom](https://www.radix-ui.com/colors/custom) page where we can generate a palette based on official radix colors but couldn't find the source for this generator. I would like to programmatically generate this inside my design system and don't want to go the website to generate than copy paste the css (_which I'm also generating programmatically_). So, a `@radix-ui/colors/customPalette` package would be really nice and will make this more useful.

**Comment by 2nPlusOne** [comment]
I would also like to see this. Open sourcing this tool could also help improve palette generator tools such as ColorBox, tints.dev, and huetone.ardov.me.

**Comment by 2nPlusOne** [comment]
@lumosminima turns out the [source](https://github.com/radix-ui/website/blob/main/pages/colors/custom.tsx) is open, although you'd have to build your own tool based on it to do what you mentioned.

**Comment by mhmdjaw** [comment]
I wish they would ship simple functions like so:

```ts
import { generateLightAccent } from '@radix-ui/colors'

const lightAccentBlue = generateLightAccent('#3D63DD')

lightAccentBlue.light4
lightAccentBlue.lightA8

const darkAccentBlue = generateDarkAccent('#3D63DD')

// ...
```

I can work on that and add those utils to this package but it doesn't seam to be active.
